exec/.gitignore

```diff
@@ -0,0 +1,7 @@
+# git does not allow empty directories.
```

Comment:
Why do you need the 'exec' directory to exist? It is automatically created during the cmake build.

Reply:
@GeorgeGayno-NOAA, The only reason we add this empty exec dir in the repository is that one can know the directory structure (exec dir) before building the package. But, anyway, I have deleted this exec/.gitignore file in this PR. 

P.S., another advantage of having this exec/.gitignore file is prevent someone incidentally adding/committing those exec/executable.x files back to the repository.